src/app/inventory/store/sockets.ts

```diff
@@ -649,32 +624,52 @@ function buildSocket(
     if (reusablePlugs) {
       // Get options from live info
       for (const reusablePlug of reusablePlugs) {
-        const built = buildPlug(defs, reusablePlug, socketDef, plugObjectivesData, plugSet);
-        addPlugOption(built, plugged, plugOptions);
+        if (plugged && reusablePlug.plugItemHash === plugged.plugDef.hash) {
+          plugOptions.push(plugged);
+        } else {
+          const built = buildPlug(defs, reusablePlug, plugObjectivesData, plugSet);
+          if (built && filterReusablePlug(built)) {
+            plugOptions.push(built);
+          }
+        }
```

Comment:
I replaced `addPlugOption` so that we could entirely avoid building the redundant plug just to throw it away, and so we could avoid the `shift`